Source/NSTextView.m

```diff
@@ -6078,6 +6080,116 @@ - (void) performFindPanelAction: (id)sender
 		withTextView: self];
 }
 
+// NSTextFinder methods implementation...
+// isSelectable, isEditable, string, selectedRanges, setSelectedRanges, replaceCharactersInRange:withString:
+// implemented by NSTextView already...
+
+- (BOOL) allowsMultipleSelection
+{
+  return NO;
+}
+
+- (NSString *) stringAtIndex: (NSUInteger)characterIndex
+              effectiveRange: (NSRangePointer)outRange
+      endsWithSearchBoundary: (BOOL *)outFlag
+{
+  return [self string];
+}
+
+- (NSUInteger) stringLength
+{
+  return [[self string] length];
+}
+
+- (NSRange) firstSelectedRange
+{
+  NSValue *r = [[self selectedRanges] objectAtIndex: 0];
+  return [r rangeValue];
+}
+
+- (BOOL) shouldReplaceCharactersInRanges: (NSArray *)ranges withStrings: (NSArray *)strings
+{
+  NSUInteger idx = 0;
+  FOR_IN(NSValue*, rv, ranges)
+    {
+      NSRange r = [rv rangeValue];
+      NSString *str = [strings objectAtIndex: idx];
+      if (![self shouldChangeTextInRange: r replacementString: str])
+        {
+          return NO;
+        }
+      idx++;
+    }
+  END_FOR_IN(ranges);
+  return YES;
+}
+
+- (void) didReplaceCharacters
+{
+  [self didChangeText];
+}
+
+- (NSView *) contentViewAtIndex: (NSUInteger)index effectiveCharacterRange: (NSRangePointer)outRange
+{
+  return nil;
+}
+
+- (NSArray *) rectsForCharacterRange: (NSRange)range
+{
+  NSUInteger rectCount = 0;
+  NSRect *rects;
+  NSMutableArray *result = [NSMutableArray array];
+  
+  if (_layoutManager)
```

Comment:
Why this check? In all other places we expect that the layout manager exists, and of course it does.

Reply:
Removed.